app/ide-desktop/lib/client/src/bin/server.ts

```diff
@@ -78,6 +80,18 @@ async function findPort(port: number): Promise<number> {
     return await portfinder.getPortPromise({ port, startPort: port })
 }
 
+// ================
+// === Port Use ===
+// ================
+
+/**
+ * Check if a port is in use.
+ */
+async function isPortInUse(port: number): Promise<boolean> {
+    const nearestPort = await findPort(port)
```

Comment:
we use this approach because (for some weird reason) approach with starting a server on the port doesn't work well(it returns ok even if the port is busy)

Reply:
we might want to limit the amount of ports checked? not sure how difficult that is

Reply:
Yeah, it's possible. Good catch